constantine/eth_verkle_ipa/common_utils.nim

```diff
@@ -101,15 +102,20 @@ func foldScalars*(res: var openArray[Fr[Banderwagon]], a,b : View[Fr[Banderwagon
     bx.prod(b[i], x)
     res[i].sum(a[i], bx)
 
-func foldPoints*(res: var openArray[EC_P], a,b : View[EC_P], x: Fr[Banderwagon])=
+func foldPoints*(res: var openArray[ECP_TwEdwards_Aff[Fp[Banderwagon]]], a,b : StridedView[ECP_TwEdwards_Aff[Fp[Banderwagon]]], x: Fr[Banderwagon])=
   ## Computes res[i] = a[i] + b[i] * x
   doAssert a.len == b.len , "Should have equal lengths!"
 
+  # TODO extra Nim alloc to refactor away
+  var t = newSeq[ECP_TwEdwards_Prj[Fp[Banderwagon]]](res.len)
+
   for i in 0 ..< a.len:
     var bx {.noInit.}: EC_P
-    bx = b[i]
+    bx.fromAffine(b[i])
     bx.scalarMul_vartime(x)
-    res[i].sum(a[i],bx)
+    t[i].madd_vartime(bx, a[i])
+
+  res.asUnchecked().batchAffine(t.asUnchecked(), res.len)
```

Comment:
It's unclear whether we benefit from affine or projective here. To be analyzed after IPA refactoring.